### PR TITLE
Make HostInterface optional until full DI is in.

### PR DESCRIPTION
**Pull request by yarongmu-google**
# Tests

```
pytest
```

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X ] I have performed a self-review of my code.
- [X ] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X ] I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

File: tpu_commons/backend.py [hunk 1]
```diff
@@ -28,13 +30,15 @@ class TPUBackend(BackendInterface):
   and managing the concrete TPU worker instance and delegating calls to it.
   """
 
-    def __init__(self, host_interface: HostInterface, **worker_kwargs):
+    def __init__(self,
+                 host_interface: Optional[HostInterface] = None,
```
Comment (yaochengji) [inline tpu_commons/backend.py:34]: Do we need to add a unit test when host_interface is None?

Comment (yarongmu-google) [reply tpu_commons/backend.py:34]: I wouldn't worry about it ... it is there because a vllm PR needs to be submitted soon. Once that one is on, this PR will be reverted :)

But I can if you insist - WDYT?

Comment (yaochengji) [reply tpu_commons/backend.py:34]: I don't think the test is needed if it will be reverted soon. Thanks!
